app/src/main/java/com/github/se/cyrcle/ui/theme/atoms/Button.kt

```diff
@@ -156,7 +158,36 @@ fun Button(
   Button(
       onClick = { if (!disabled.value) onClick() },
       modifier = modifier.testTag(testTag),
-      colors = getButtonColors(colorLevel)) {
+      colors = getButtonColors(colorLevel),
+      enabled = !disabled.value) {
+        Text(text, Modifier.testTag("${testTag}Text"))
+      }
+}
+
+/**
+ * Create a themed toggle button, with simplified arguments.
+ *
+ * @param value The mutable boolean that this toggle represent and modify.
+ * @param modifier Chained modifier. `.testTag` will be overwritten, use the `testTag` for this.
+ * @param colorLevel The color scheme of the object.
+ * @param testTag The test tag of the object.
+ */
+@Composable
+fun ToggleButton(
+    text: String,
+    value: MutableState<Boolean>,
+    modifier: Modifier = Modifier,
+    colorLevel: ColorLevel = ColorLevel.PRIMARY,
+    testTag: String = "ToggleButton"
+) {
+  Button(
+      onClick = { value.value = !value.value },
```

Comment:
I think this is too restrictive. Specifically, it is not always the case that this button is going to get called with a mutable state of boolean. As an example, in the case of filter, I'd like it to act on a set. I think it is imperative to have a customazible onclick function

Reply:
The functionality has been added in latest commit ^^